chain/state/statetree.go

```diff
@@ -175,17 +236,13 @@ func (st *StateTree) Snapshot(ctx context.Context) error {
 	ctx, span := trace.StartSpan(ctx, "stateTree.SnapShot")
 	defer span.End()
 
-	ss, err := st.Flush(ctx)
-	if err != nil {
-		return err
-	}
+	st.snaps.addLayer()
```

Comment:
Why is this needed?

Reply:
Snapshots add a layer to the snapshot set, that way any changes after we call snapshot go onto their own layer